### PR TITLE
Allow absolute uri in authentication properties for cookie auth redirect

### PR DESCRIPTION
**Pull request by troydai**
/cc @PinpointTownes @Tratcher

### CODE REVIEW

**Comment by PinpointTownes** [comment]
Nice, thanks! :clap: 

Question: is it worth logging an info/warning message when a returnUrl query string parameter is rejected?

(minor nit: you could remove the variable declaration at line 317 and use `var` at line 325)

File: src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs [hunk 1]
```diff
@@ -314,13 +313,11 @@ private CookieOptions BuildCookieOptions()
...
-
                 // set redirect uri in order:
                 // 1. properties.RedirectUri
                 // 2. query parameter ReturnUrlParameter
                 var redirectUri = properties.RedirectUri;
-                if (string.IsNullOrEmpty(redirectUri) || !IsHostRelative(redirectUri))
+                if (string.IsNullOrEmpty(redirectUri))
                 {
                     redirectUri = Request.Query[Options.ReturnUrlParameter];
                     if (string.IsNullOrEmpty(redirectUri) || !IsHostRelative(redirectUri))
```
Comment (Tratcher) [inline src/Microsoft.AspNetCore.Authentication.Cookies/CookieAuthenticationHandler.cs]: Please comment why we require these to be relative.

**Comment by Tratcher** [comment]
:shipit:

**Comment by troydai** [comment]
Comments are added, waiting for the CI to pass.

**Pull request merged**